src/modules/dashboard/sagas/actions.ts

```diff
@@ -241,10 +242,108 @@ function* createMoveFundsAction({
   }
 }
 
+function* createMintTokensAction({
+  payload: { colonyAddress, colonyName, nativeTokenAddress, amount },
+  meta: { id: metaId, history },
+  meta,
+}: Action<ActionTypes.COLONY_ACTION_MINT_TOKENS>) {
+  let txChannel;
+  try {
+    const apolloClient = TEMP_getContext(ContextModule.ApolloClient);
+
+    if (!amount) {
+      throw new Error('Amount to mint not set for mintTokens transaction');
+    }
+
+    txChannel = yield call(getTxChannel, metaId);
+
+    // setup batch ids and channels
+    const batchKey = 'mintTokens';
+    const mintTokens = {
+      id: `${metaId}-mintTokens`,
+      channel: yield call(getTxChannel, `${metaId}-mintTokens`),
+    };
+    const claimColonyFunds = {
+      id: `${metaId}-claimColonyFunds`,
+      channel: yield call(getTxChannel, `${metaId}-claimColonyFunds`),
+    };
+
+    // create transactions
+    yield fork(createTransaction, mintTokens.id, {
+      context: ClientType.ColonyClient,
+      methodName: 'mintTokens',
+      identifier: colonyAddress,
+      params: [amount],
+      group: {
+        key: batchKey,
+        id: meta.id,
```

Comment:
You have `metaId` already destructured, why are you still using the `meta.id` value from the `meta` object ?